keps/sig-storage/20190130-csi-raw-block-support.md

```diff
@@ -0,0 +1,110 @@
+---
+title: CSI Raw Block Volumes
+authors:
+  - "@bswartz"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-storage
+reviewers:
+  - "@msau42"
+  - "@saad-ali"
+approvers:
+  - name: "@saad-ali"
+editor: TBD
+creation-date: 2019-01-30
+last-updated: 2019-01-31
+status: implementable
+---
+
+# CSI Raw Block Volumes
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Proposal](#proposal)
+* [Test Plan](#test-plan)
+   * [Unit tests](#unit-tests)
+   * [E2E tests](#e2e-tests)
+* [Graduation Criteria](#graduation-criteria)
+   * [Alpha -&gt; Beta](#alpha---beta)
+   * [Beta -&gt; GA](#beta---ga)
+* [Implementation History](#implementation-history)
+   * [K8s 1.12: Alpha](#k8s-112-alpha)
+   * [K8s 1.13: Alpha](#k8s-113-alpha)
+
+## Summary
+
+Raw block support has been added to both the core of Kubernetes (including
+API support and support in kubelet for a subset of volume types) as well as
+the CSI specification. This feature is specifically to add raw block support
+to kubelet for volumes of the "csi" type by taking advantage of the
+now-standardized CSI RPCs.
+
+https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/raw-block-pv.md
+
+## Motivation
+
+The future of persistent volume support should be mostly CSI-type volumes,
+with a few notable exceptions. It's therefore essential that CSI volumes
+can implement all of the features that Kubernetes has, including support
+for raw block volumes.
+
+### Goals
+
+CSI plugins should be able to create, publish, and attach raw block volumes
+in accordance with the raw block support APIs.
+
+### Non-Goals
+
+Raw block support added in any particular CSI driver.
+
+## Proposal
+
+Modify the CSI volume package of kubelet to declare support for raw block
+volumes, and add the CSI GRPC client code to call the appropriate node RPCs
+for raw block volumes when attaching pods to volumes of that type. This
```

Comment:
Could you add a few lines about:

1. What CSI calls are called for raw block and by whom (kubelet) and when (as part of `MountDevice` or whatever operation).
2. What preparation, if any, kubelet does (e.g. create a folder or symlink, etc.).